src/types/rule.ts

```diff
@@ -1,7 +1,24 @@
 import { DiagnosticSeverity } from '@stoplight/types';
-import { RuleFunction, RuleType } from './enums';
-
-export type Rule = IRule | TruthyRule | XorRule | LengthRule | AlphaRule | PatternRule | CasingRule | SchemaRule;
+import { AlphaRule } from '../functions/alphabetical';
+import { CasingRule } from '../functions/casing';
+import { LengthRule } from '../functions/length';
+import { PatternRule } from '../functions/pattern';
+import { SchemaRule } from '../functions/schema';
+import { SchemaPathRule } from '../functions/schema-path';
+import { TruthyRule } from '../functions/truthy';
+import { XorRule } from '../functions/xor';
+import { RuleType } from './enums';
+
+export type Rule =
```

Comment:
@philsturgeon in #884 you were hinting that we may not need this any longer ("The whole export type Rule thing seems outdated now no?").

To be honest, I don't know. Yet.

`Rule` seems to be quit used throughout the code base and the tests.

It looks like replacing it by `IRule` might do the trick, but I haven't tried that yet.

Reply:
AFAIK this is something we used to do when there was no custom functions. Now, I think we should try to make our ruleset just another ruleset, and make sure that functions are just custom functions, so we are dogfooding the system we expect others to use. This will help in the future when we want to spin out our ruleset into another module. 

If its not possible, I probably misunderstood something about how we implemented custom functions.

Reply:
We don't support custom function arguments validation.

Reply:
Some extra context
We can distinguish two kind of functions:
- the custom ones (provided by rulesets)
- the built-in ones (provided by Spectral)

What we are touching here are the built-in ones. 
They will always be loaded regardless of rulesets.

Functions placed next to oas rulesets are the ones we will want to move to another module at some point.

Reply:
Ahhh yes of course, great!